Tests/MongoKittenTests/CRUDTests.swift

```diff
@@ -1,55 +1,180 @@
-//import NIO
-//import MongoKitten
-//import XCTest
-//
-//let dbName = "test"
-//
-//class RemoteDatabaseCRUDTests : XCTestCase {
+import NIO
+import MongoKitten
+import XCTest
+
+let dbName = "KittenTest"
+
+class CrudTests : XCTestCase {
+  struct DummyAccount: Codable {
+    static let collectionName = "DummyAccounts"
+
+    let _id: ObjectId
+    var name: String
+    var password: String
+    var age: Int
+    
+    init (Name name: String, Password password: String, Age age: Int) {
+      self._id = ObjectId()
+      self.name = name
+      self.password = password
+      self.age = age
+    }
+  }
+  
+  let settings = ConnectionSettings(
+    authentication: .auto(username: "admin", password: "Autimatisering1"),
+    authenticationSource: "admin",
+    hosts: [
+        .init(hostname: "localhost", port: 27017)
+    ],
+    targetDatabase: dbName,
+    applicationName: "Test MK6"
+  )
+  var mongo : MongoDatabase!
+
+  override func setUpWithError() throws {
+    mongo = try MongoDatabase.synchronousConnect(settings: settings)
+  }
+
+  override func tearDownWithError() throws {
+    try mongo.drop().wait()
+  }
+
+  func testCreateDummyAccount () throws {
+    let schema = mongo[DummyAccount.collectionName]
+    let startingCount = try schema.count().wait()
+
+    let dummyAccount = DummyAccount(Name: "Dum", Password: "my", Age: 69)
+    let _ = try schema.insertEncoded(dummyAccount).wait()
+    XCTAssertEqual(try schema.count().wait(), startingCount+1)
+  }
+
+  func testBulkCreateDummyAccounts () throws { 
+    struct SetupFailed: Error {}
+    let schema = mongo[DummyAccount.collectionName]
+    let startingCount = try schema.count().wait()
+
+    let dummyAccounts = [
+      DummyAccount(Name: "Test", Password: "ing", Age: 77),
+      DummyAccount(Name: "To", Password: "see", Age: 8),
+      DummyAccount(Name: "If", Password: "bulk", Age: 10),
+      DummyAccount(Name: "Inserts", Password: "will", Age: 15),
+      DummyAccount(Name: "Work", Password: "as", Age: 19),
+      DummyAccount(Name: "I", Password: "expect", Age: 30),
+      DummyAccount(Name: "Them", Password: "to", Age: 82),
+    ]
+    _ = schema.insertManyEncoded(dummyAccounts)
+    XCTAssertEqual(try schema.count().wait(), startingCount+7)
+  }
+
+  func readDummyAccount (name : String) throws -> DummyAccount? {
+    let schema = mongo[DummyAccount.collectionName]
+    if let dummy = try schema.findOne("name" == name, as: DummyAccount.self).wait() {
+      return dummy
+    }
+    return nil
+  }
+
+  func printDummyAccounts () throws {
+    let schema = mongo[DummyAccount.collectionName]
+    try schema.find().forEach{ dummy in 
+      print("\(dummy["_id"]!): Name - \(dummy["name"]!) | Password - \(dummy["password"]!) | Age - \(dummy["age"]!)")
```

Comment:
Force unwraps!

Reply:
Completely true, but 
- the alternative of having to do a if-let would be a pain for this many instances.
- this is a debug tool for the tests, not a test itself and not used it in any tests after confirming the test works

Can I per chance decode the document to a dummy instance? it'd solve this entirely.